src/client/src/apps/SimpleLauncher/spotlight/components/useSpotPrice.ts

```diff
@@ -0,0 +1,32 @@
+import { useEffect, useState } from 'react'
+import { SpotPriceTick } from 'apps/MainRoute'
+import { usePriceService } from './usePriceService'
+
+export const useSpotPrice = (currencyPair: string | undefined) => {
+  const priceService = usePriceService()
+  const [spotPrice, setSpotPrice] = useState<SpotPriceTick>()
+
+  useEffect(() => {
+    setSpotPrice(undefined)
+  }, [currencyPair, setSpotPrice])
+
+  useEffect(() => {
+    if (!priceService || !currencyPair) {
+      return
+    }
+
+    const subscription = priceService
+      .getSpotPriceStream({ symbol: currencyPair })
```

Comment:
Any reason you can't just take(1)?  That would automatically cleanup the subscription each time the effect runs.  You wouldn't need worry about giving it a handle and manually unsubscribing on unmount.

Reply:
We need a streaming price, not just the first price.